.gitlab-ci.yml

```diff
@@ -1,182 +1,206 @@
 # The testing matrix considers ruby/puppet versions supported by SIMP and PE:
 #
-# https://puppet.com/docs/pe/2017.3/overview/component_versions_in_recent_pe_releases.html
+# https://puppet.com/docs/pe/2018.1/component_versions_in_recent_pe_releases.html
 # https://puppet.com/misc/puppet-enterprise-lifecycle
-# https://puppet.com/docs/pe/2017.3/overview/getting_support_for_pe.html#standard-releases-and-long-term-support-releases
+# https://puppet.com/docs/pe/2018.1/overview/getting_support_for_pe.html
 # ------------------------------------------------------------------------------
-#  release    pup   ruby      eol
-# PE 2016.4   4.7   2.1.9  2018-10  (LTS)
-# SIMP6.0.0   4.8   2.1.9  TBD
-# PE 2017.2   4.10  2.1.9  2018-02-21
-# PE 2017.3   5.3   2.4.1  2018-07
-# PE 2018.1   ???   ?????  ????-??  (LTS)
+# Release       Puppet   Ruby   EOL
+# PE 2016.4.15  4.10     2.1.9  2018-12-31 (LTS)
+# SIMP 6.2      4.10     2.1.9  TBD
+# PE 2016.4.15  4.10     2.1.9  2018-12 (LTS)
+# PE 2017.3     5.3      2.4.4  2018-12-31
+# SIMP 6.3      5.5      2.4.4  TBD***
+# PE 2018.1     5.5      2.4.4  2020-05 (LTS)***
+#
+# *** = Modules created for SIMP 6.3+ are not required to support Puppet < 5.5
 ---
-.cache_bundler: &cache_bundler
+stages:
+  - 'sanity'
+  - 'validation'
+  - 'acceptance'
+  - 'compliance'
+  - 'deployment'
+
+image: 'ruby:2.4'
+
+variables:
+  PUPPET_VERSION:    'UNDEFINED' # <- Matrixed jobs MUST override this (or fail)
+  BUNDLER_VERSION:   '1.16.1'
+
+  # Force dependencies into a path the gitlab-runner user can write to.
+  # (This avoids some failures on Runners with misconfigured ruby environments.)
+  GEM_HOME:          .vendor/gem_install
+  BUNDLE_CACHE_PATH: .vendor/bundle
+  BUNDLE_PATH:       .vendor/bundle
+  BUNDLE_BIN:        .vendor/gem_install/bin
+  BUNDLE_NO_PRUNE:   'true'
+
+# bundler dependencies and caching
+#
+# - Cache bundler gems between pipelines foreach Ruby version
+# - Try to use cached and local resources before downloading dependencies
+# --------------------------------------
+.setup_bundler_env: &setup_bundler_env
   cache:
     untracked: true
-    # A broad attempt at caching between runs (ala Travis CI)
-    key: "${CI_PROJECT_NAMESPACE}__bundler"
+    key: "ruby-${MATRIX_RUBY_VERSION}_bundler"
```

Comment:
Why was CI_PROJECT_NAMESPACE removed from the key?

Reply:
Because we now know that [the way GitLab CI handles caching storage](https://docs.gitlab.com/ce/ci/caching/#where-the-caches-are-stored) makes using a very broad variable like `${CI_PROJECT_NAMESPACE}` pointless. 

For shell executors, the cache is always saved to **`/home/gitlab-runner/cache/<user>/<project>/<cache-key>/cache.zip`**.  The cache is already compartmentalized under `<user>/` and `<project>/` before the cache key is considered, meaning it can't be shared across projects—no matter how broad a variable is used.

There's no harm in keeping `${CI_PROJECT_NAMESPACE}`, but it also does no good.